src/dotnet/GeneratorAgent/src/Azure.Tools.GeneratorAgent/Configuration/ToolConfiguration.cs

```diff
@@ -0,0 +1,44 @@
+using System.Reflection;
+using Azure.Tools.GeneratorAgent.Configuration;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Azure.Tools.GeneratorAgent;
+
+internal class ToolConfiguration
+{
+    private readonly IConfiguration _configuration;
+    private readonly string ToolDirectory;
+    public ToolConfiguration()
+    {
+        ToolDirectory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)
+            ?? throw new InvalidOperationException("Unable to determine tool installation directory");
+
+        _configuration = CreateConfigurationInternal();
+    }
+
+    public IConfiguration Configuration => _configuration;
```

Comment:
```suggestion
    public virtual IConfiguration Configuration => _configuration;
```

We don't need to do this now, we can hold back until we need to mock for testing.   But we'll want to mark the important stuff as `virtual` so that we can use `Moq` or create our own stubs to control behavior.

I'll call out opportunities in this class, but the concept applies broadly.   Feel free to punt these to a future PR when we need them.